### DISCUSSION TITLE
How to print the 5-digit icons

### DISCUSSION THREAD

**Discussion by andrasnemethcg** [ryanoasis/nerd-fonts#1077] [Q&A] [answered]
I would like to print 'nf-md-kubernetes', hex code: f10fe to the terminal, from bash. 
The FAQ recommended way `echo $'\ue62b'` works for 4-digit codes, but not for the 5-digit ones. 
How do I print to bash terminal the icon with hex code: f10fe?
Thanks!

**Comment by Finii** [comment]
Sorry, can you point me to the exact location where that is mentioned in the FAQ (which FAQ)?

You would probably need to use UTF16, which has an open ticket already #1059 

Usually on Linux I just enter the glyph itself into the `echo` (making that an UTF8 or whatever document) instead of putting the code into the `echo` instruction which is semi-portable anyway ;-)

`ctrl`+`shift`+`u` `f` `1` `0` `f` `e` `enter` works on my machine at least in all applications.

**Reply by andrasnemethcg** [reply]
Here is the [link in FAQ](https://github.com/ryanoasis/nerd-fonts/wiki/FAQ-and-Troubleshooting#how-do-i-use-glyphs-in-my-terminal).

I need to print it from bash script.

**Reply by Finii** [reply]
Thanks for the link; I will change that once we have a solution for #1059.

**Reply by Finii** [reply]
UTF8 with raw codes (`\x`) or the `\U` (capital U) thing might be your friend? See below.

**Reply by FernandoBasso** [reply]
Just in case it may save others some head-scratching, I think `Ctrl`+`Shift`+`u` is a Gtk thing. I don't think it works on QT apps or KDE in general.

**Comment by Finii** [comment]
![image](https://user-images.githubusercontent.com/16012374/214532664-0d05b87b-87ac-4414-9d8d-3cc82d7ddb18.png)

Paste glyph into https://onlineunicodetools.com/convert-unicode-to-utf16

![image](https://user-images.githubusercontent.com/16012374/214532870-ab5b0d1e-7e3a-4afb-9498-72ccca12f6e4.png)

use UTF16 codes like `echo "\udb84\udcfe"`

:warning: This does not work on my machine with `zsh`, probably also not on yours.

Use instead UTF8...: https://onlineunicodetools.com/convert-unicode-to-utf8

![image](https://user-images.githubusercontent.com/16012374/214535516-07ba95e6-6578-476a-be6e-dce2daf4e135.png)

`echo "\xf3\xb1\x83\xbe"`

![image](https://user-images.githubusercontent.com/16012374/214535565-ed0258c7-713f-4d07-98be-2ec7460945d2.png)

_Edit: Add UTF8 variant_

**Comment by Finii** [comment, marked as answer]
If you are willing to use `printf` instead of `echo` (which you should anyway, because its behavior is far more portable) you can use the `\U` instead of the `\u`:

![image](https://user-images.githubusercontent.com/16012374/214534129-1cb365e6-c540-4781-b908-73e594d9190e.png)

**Reply by Finii** [reply]
Btw, `zsh`'s echo also has `\U`.

Both are shell builtins, do not know about `bash`.

**Reply by andrasnemethcg** [reply]
Thanks @Finii 
This works fine!
